### PR TITLE
refactor: pull out delta CMD node

### PR DESCRIPTION
**Pull request by jcace**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ func main() {
 		Name:        "delta",
 		Description: "A deal making engine microservice for the filecoin network",
 		Version:     fmt.Sprintf("%s+git.%s\n", cfg.Common.Version, cfg.Common.Commit),
+		Flags:       cmd.CLIConnectFlags,
```
Comment (jcace) [inline main.go:49]: this has to be here otherwise it will complain that the flags aren't defined, if they're passed in

File: cmd/deltaCmdNode.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+// Verify that DDM API is reachable
+func healthCheck(url string, authKey string) error {
+	req, err := http.NewRequest("GET", url+"/open/node/info", nil)
+	if err != nil {
+		return fmt.Errorf("could not construct http request %v", err)
+	}
+
+	req.Header.Set("Authorization", "Bearer "+authKey)
```
Comment (jcace) [inline cmd/deltaCmdNode.go:68]: that endpoint doesn't actually require auth - should we use a different one to validate it?

File: cmd/deltaCmdNode.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		}
+	}
+
+	deltaAuth := c.String("delta-auth")
+	if deltaAuth == "" {
+		deltaAuth = os.Getenv("DELTA_AUTH")
+		if deltaAuth == "" {
+			return nil, fmt.Errorf("DELTA_AUTH env variable or --delta-auth flag is required")
+		}
+	}
```
Comment (jcace) [inline cmd/deltaCmdNode.go:47]: i'm not a fan of this logic - wonder if there's a cleaner way of doing this?

essentially - take the value from cli - if undefined, take the value from env. if undefined - use default, or error.

Comment (alvin-reyes) [reply cmd/deltaCmdNode.go:47]: I created a `getFlagOrEnvVar` function that will wrap the repetitive value check.

```
func getFlagOrEnvVar(c *cli.Context, flagName, envVarName, defaultValue string) string {
	value := c.String(flagName)
	if value == "" {
		value = os.Getenv(envVarName)
		if value == "" {
			value = defaultValue
		}
	}
	return value
}
```

I then call it from the node initialization

```
deltaApi := getFlagOrEnvVar(c, "delta-api", "DELTA_API", "http://localhost:1414")
deltaAuth := getFlagOrEnvVar(c, "delta-auth", "DELTA_AUTH", "")

```

File: cmd/deltaCmdNode.go (hunk 2 above)
Comment (alvin-reyes) [reply cmd/deltaCmdNode.go:68]: I added two health check endpoints

```
/health/check/ping - unauthenticated health check.
/health/check/auth/ping - authenticated health check i.e check if the API key is valid
```

**Review by alvin-reyes — APPROVED** [review]
LGTM.

Tested this on my local and it works. If you need to add more to this PR, just commit and we can discuss. If not, I'll merge this and include it on v1.0.1.

Thank you!

**Pull request merged**